ckanext/canada/logic.py

```diff
@@ -442,44 +446,70 @@ def canada_job_list(up_func, context, data_dict):
     job_list = up_func(context, data_dict)
 
     for job in job_list:
+        job_args = None
+        job_kwargs = None
         try:
             job_obj = Job.fetch(job.get('id'))
-            job_args = job_obj.args[0]
+            if job_obj.args:
+                job_args = job_obj.args[0]
+            if job_obj.kwargs:
+                job_kwargs = job_obj.kwargs
         except (NoSuchJobError, KeyError):
             continue
 
         job_title = _(job.get('title', 'Unknown Job'))
+        rid = None
+        gid = None
 
         if job_obj.func_name in JOB_MAPPING:
-            rid = JOB_MAPPING[job_obj.func_name]['rid'](job_args)
+            if job_args:
+                rid = JOB_MAPPING[job_obj.func_name]['rid'](job_args)
+            if job_kwargs:
+                gid = JOB_MAPPING[job_obj.func_name]['gid'](job_kwargs)
             icon = JOB_MAPPING[job_obj.func_name]['icon']
         else:
-            rid = None
             job_title = _('Unknown Job')
             icon = 'fa-circle-o-notch'
 
         job_info = {}
-        if rid:
+        if rid or gid:
             current_user = get_action('get_site_user')({'ignore_auth': True}, {})['name']
             if has_request_context():
                 try:
                     current_user = g.user
                 except (TypeError, AttributeError):
                     pass
-            try:
-                resource = get_action('resource_show')({'user': current_user}, {'id': rid})
-            except (ObjectNotFound, NotAuthorized):
-                continue
-            job_info = {'name_translated': resource.get('name_translated'),
-                        'resource_id': rid,
-                        'url': h.url_for('dataset_resource.read',
-                                         id=resource.get('package_id'),
-                                         resource_id=rid)}
+            if rid:
+                try:
+                    resource = get_action('resource_show')({'user': current_user}, {'id': rid})
+                except (ObjectNotFound, NotAuthorized):
+                    continue
+                job_info = {'name_translated': resource.get('name_translated'),
+                            'resource_id': rid,
+                            'url': h.url_for('dataset_resource.read',
+                                            id=resource.get('package_id'),
+                                            resource_id=rid)}
+            if gid:
+                if gid == config.get('ckan.site_id'):
+                    job_info = {'name': _('Entire Site')}
+                else:
+                    try:
```

Comment:
using a `'reindex_type'` key for group/organization in the job would mean not having to try/except group/organization here too

Reply:
we actually don't have any groups in our site, and the capability is fully shut off, so this is to just cover some 500 error bases in the view really. And as this is just an admin view that is not really used much at all, it could just be fine.

Unless if you think that having a `reindex_type` key in core would be beneficial to core?